nixos/modules/services/web-apps/phylactery.nix

```diff
@@ -0,0 +1,44 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let cfg = config.services.phylactery;
+in {
+  options.services.phylactery = {
+    enable = mkEnableOption "Whether to enable Phylactery server.";
+
+    host = mkOption {
+      type = types.str;
+      default = "";
+      description = "Listen host for Phylactery.";
+    };
+
+    port = mkOption {
+      type = types.port;
+      description = "Listen port for Phylactery.";
+    };
+
+    library = mkOption {
+      type = types.str;
+      description = "Path to CBZ library.";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.phylactery = {
+      environment = {
+        PHYLACTERY_ADDRESS = "${cfg.host}:${toString cfg.port}";
+        PHYLACTERY_LIBRARY = cfg.library;
+      };
+
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        ConditionPathExists = cfg.library;
+        DynamicUser = true;
+        ExecStart = "${pkgs.phylactery}/bin/phylactery";
```

Comment:
Adding a `package` option with this as the default would be nice.

Reply:
Could you please elaborate on the reason?

Reply:
Well, if I wanted to override `pkgs.phylactery` I'd have to use a nixpkgs overlay, and that can effect other unrelated usages of `phylactery`. If instead I could set `config.services.phylactery.package` that would make it contained to `nixos/phylactery`.

Reply:
Thanks, done!